src/event.rs

```diff
@@ -2,8 +2,7 @@ use std::{net::SocketAddr, sync::Arc};
 
 use crate::Error;
 
-#[derive(Clone, Copy)]
-#[non_exhaustive]
```

Comment:
I don't think we'll be introducing any more event variants than necessary. Introducing the event listener callback works well for handling errors that occur outside the request/response cycle and also for the "server ready" callback. However, I believe it's best to use this pattern sparingly in both the private and public API of the framework. I'm open to suggestions if anyone thinks this might result in avoidable breaking changes to the public API in the future.